extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -197,10 +236,43 @@ public void emit(final Event event)
     }
   }
 
+  private void emitSegmentMetadataEvent(SegmentMetadataEvent event)
+  {
+    Map<String, Object> eventMap = event.toMap();
+    long createdTimeMs = ((DateTime) eventMap.get(SegmentMetadataEvent.CREATED_TIME)).getMillis();
+    Timestamp createdTimeTs = Timestamp.newBuilder().setSeconds(createdTimeMs / 1000)
+        .setNanos((int) ((createdTimeMs % 1000) * 1000000)).build();
+    long startTimeMs = ((DateTime) eventMap.get(SegmentMetadataEvent.START_TIME)).getMillis();
+    Timestamp startTimeTs = Timestamp.newBuilder().setSeconds(startTimeMs / 1000)
+        .setNanos((int) ((startTimeMs % 1000) * 1000000)).build();
+    long endTimeMs = ((DateTime) eventMap.get(SegmentMetadataEvent.END_TIME)).getMillis();
+    Timestamp endTimeTs = Timestamp.newBuilder().setSeconds(endTimeMs / 1000)
+        .setNanos((int) ((endTimeMs % 1000) * 1000000)).build();
+
+    DruidSegmentEvent.Builder druidSegmentEventBuilder = DruidSegmentEvent.newBuilder()
+        .setDatasource(eventMap.get(SegmentMetadataEvent.DATASOURCE).toString())
```

Comment:
Same applies to all the DateTime casting above
```
 ((DateTime) eventMap.get(SegmentMetadataEvent.END_TIME))
```

Reply:
+1 to adding getters. This would only be necessary if we had to extract those fields from a generic `Event` object, but since we already deal with `SegmentMetadataEvent` directly here we can call the getters instead to make this less brittle.